### ISSUE TITLE
Plugin Idea: Packt Free Ebook listings notifications

### ISSUE THREAD

**Issue by daredevil82** [pyslackers/sirbot-plugins#10]
The publisher Packt has a [free ebooks page](https://www.packtpub.com/packt/offers/free-learning) that updates every 24 hours or so, and a good portion of them are related to channel topics and Python in general.  We could define a list of criteria for a book to match, and plugin to check the current day's offering.  If the book matches the criteria, it would then post a link in the #general channel.

An additional enhancement would be to check the Amazon API for the user review count and stars for the book and include that in the channel posting.

**Comment by ovv** [comment]
Thanks for the idea.

Do you know if they provide an rss feed ? 
A general plugin to follow RSS feeds  would be more globally usable.

PS: I update your comment because the link was bugged

**Comment by daredevil82** [comment]
I agree a RSS feed would be easier, but I don't believe Packt offers such a stream, unfortunately.  I am signed up for email notifications from them, but it is around a twice-monthly sort of thing.  

They do have their own [subreddit](https://www.reddit.com/r/Packt/), but postings are somewhat inconsistent.

**Comment by ovv** [comment]
Email and reddit are both services I hope to integrate at one point :)

I'm not big onto scrapping but you are more than welcome if you want to give it a shot. We might need #9 first.

**Comment by daredevil82** [comment]
Yes,  since this would be a scheduled task.  I'd need to learn the slackbot API first, though.

Right now, this is just the idea.  I'd be able to to the scraping code, but the slack integration is a big ❓❓❓ for me.

**Comment by ovv** [comment]
Well I'm trying to simplify as much as possible the slack integration.

You can look [here](https://github.com/pyslackers/sirbot-pythondev/tree/master/sirbot/pythondev/slack) and [here](https://github.com/pyslackers/sirbot-pythondev/blob/master/sirbot/pythondev/github.py) for some example of how the messages are built with sirbot.
